clang/test/dpct/syncthreads.cu

```diff
@@ -445,3 +448,16 @@ __global__ void test20() {
   recursive(10);
   __syncthreads();
 }
+
+__global__ void test21(float *ptr1, float *ptr2, int step1, int step2) {
+  int idx1 = blockDim.x * blockIdx.x + threadIdx.x;
+  int idx2 = blockDim.x * blockIdx.x + threadIdx.x;
+  for (int i = 0; i < 10; i++) {
+    ptr2[idx1] = 123;
+    idx1 += step1;
+    // CHECK: (item_ct1.get_local_range(2) < std::min(step1, step2)) ? item_ct1.barrier(sycl::access::fence_space::local_space) : item_ct1.barrier();
```

Comment:
Please comment the the logic behind the condition: patten.

Reply:
```
local_range(2) loop0     loop1    loop2    ... loopn
     0         mem[0]    mem[s]   mem[2s]      mem[ns]
     1         mem[1+0]  mem[1+s] mem[1+2s]    mem[1+ns]
     2         mem[2+0]  mem[2+s] mem[2+2s]    mem[2+ns]
    ...
     m         mem[m+0]  mem[m+s] mem[m+2s]    mem[m+ns]
```
We can make sure that there is no overlap in the same iteration since idx should equal to `local_id(2) + C`.
Next, we need to make sure there is no overlap among iterations.
The memory range in an iteration is `local_range(2)`, then if `s > local_range(2)`, the next iteration start point is larger than previous end, so there is no overlap.